core/service.go

```diff
@@ -323,21 +329,25 @@ func (s *Service) handleReceivedMessage(msg network.Message) (err error) {
 // announce messages (block announce messages include the header but the full
 // block is required to execute `core_execute_block`).
 func (s *Service) ProcessBlockAnnounceMessage(msg network.Message) error {
+	data := msg.(*network.BlockAnnounceMessage)
 
-	// TODO: check if we should send block request message
+	header, err := types.NewHeader(data.ParentHash, data.Number, data.StateRoot, data.ExtrinsicsRoot, data.Digest)
+	if err != nil {
+		return err
+	}
 
-	// TODO: update message properties and use generated id
-	blockRequest := &network.BlockRequestMessage{
-		ID:            1,
-		RequestedData: 2,
-		StartingBlock: []byte{},
-		EndBlockHash:  optional.NewHash(true, common.Hash{}),
-		Direction:     1,
-		Max:           optional.NewUint32(false, 0),
+	// TODO: check if we should send block request message
+	_, err = s.blockState.GetHeader(header.Hash())
+	if err != nil && err.Error() == "Key not found" {
```

Comment:
This looks like a "magic" string, shall it be a const and shared among GetHeader and places where the check is required ?

Reply:
the error actually comes from here: https://github.com/dgraph-io/badger/blob/f5b63211d7f3e2f5f8b698893313b2a54e4df7de/errors.go#L36

I don't think we are using it anywhere else at the moment, if I wanted to do an error check I would need to import badger into this package. I think it should be imported somewhere else in gossamer, perhaps state, if we are going to use this often